cohorts/hash.py

```diff
@@ -0,0 +1,42 @@
+import copy
+
+# courtesy of https://stackoverflow.com/a/8714242/3457743
```

Comment:
If you like. I don't have a strong preference re: more or fewer files, except when it helps organization.